### ISSUE TITLE
Attempting to attach an unknown Portal type. BasePortalOutlet accepts either a ComponentPortal or a TemplatePortal.

### ISSUE THREAD

**Issue by MardariG** [DanielYKPan/date-time-picker#369]
Just got this error in production.

```
"dependencies": {
"ng-pick-datetime": "^6.0.4",
"@angular/cdk": "^6.1.0",
"@angular/material": "^6.1.0",
"@angular/core": "6.0.3",
........

"devDependencies": {
    "@angular/cli": "6.0.3",
    "@angular/compiler-cli": "6.0.3",
```
this is how my code looks:

```
<mat-form-field>
           <input (change)="calculateTotal()" (dateTimeChange)="calculateTotal();"
                       matInput placeholder="Date Time To:"
                       [owlDateTimeTrigger]="dt13"
                       [owlDateTime]="dt13"
                       [(ngModel)]="range"
                       selectMode="rangeTo">
           <owl-date-time firstDayOfWeek="6" #dt13></owl-date-time>
</mat-form-field>
```
So as you can see I use **matInput** here also, and **mat-form-field** as a wrapper, so maybe this can cause some conflicts. 
In development mode everything  works perfectly.

Here is the production error trace if it helps:

```
0.83a6145142c144fbb003.chunk.js:1 ERROR Error: Attempting to attach an unknown Portal type. BasePortalOutlet accepts either a ComponentPortal or a TemplatePortal.
    at 0.83a6145142c144fbb003.chunk.js:1
    at e.t.attach (0.83a6145142c144fbb003.chunk.js:1)
    at t.attach (0.83a6145142c144fbb003.chunk.js:1)
    at e.openAsPopup (0.83a6145142c144fbb003.chunk.js:1)
    at e.open (0.83a6145142c144fbb003.chunk.js:1)
    at t.handleClickOnHost (0.83a6145142c144fbb003.chunk.js:1)
    at Object.handleEvent (main.83a6145142c144fbb003.bundle.js:1)
    at Object.handleEvent (0.83a6145142c144fbb003.chunk.js:1)
    at Object.handleEvent (0.83a6145142c144fbb003.chunk.js:1)
    at Kr (0.83a6145142c144fbb003.chunk.js:1)
Ut @ 0.83a6145142c144fbb003.chunk.js:1
t.handleError @ 0.83a6145142c144fbb003.chunk.js:1
Kr @ 0.83a6145142c144fbb003.chunk.js:1
(anonymous) @ 0.83a6145142c144fbb003.chunk.js:1
(anonymous) @ 0.83a6145142c144fbb003.chunk.js:1
t.invokeTask @ 0.83a6145142c144fbb003.chunk.js:1
onInvokeTask @ 0.83a6145142c144fbb003.chunk.js:1
t.invokeTask @ 0.83a6145142c144fbb003.chunk.js:1
e.runTask @ 0.83a6145142c144fbb003.chunk.js:1
e.invokeTask @ 0.83a6145142c144fbb003.chunk.js:1
v @ 0.83a6145142c144fbb003.chunk.js:1
b @ 0.83a6145142c144fbb003.chunk.js:1
```

**Comment by DanielYKPan** [comment]
I tested it with a new angular cli project. And it works fine in both dev or prod environment.

You might consider other parts of your app causing this issue.
You could build and share an app to show the issue. I would close this until it is confirmed as an issue.

**Comment by MardariG** [comment]
I've used webpack until now, and now just migrated to angular CLI latest version, to test this issue, and it happens again. Just make a build with --prod and it fails when you try to open the datepicker modal.

**Comment by syamaner** [comment]
Hi @MardariG, I also ran into this issue on a project that was upgraded from Angular 5.x to 6.x. Having tested on a fresh CLI application (6.0.4), date-time-picker is working as expected. Perhaps in some cases, following the upgrade path breaks compatibility. 
I have used --prod and had AOT and buildOptimizer enabled and all is working fine.

**Comment by MardariG** [comment]
I spend too much time to understand what caused the issue, because it is a production only issue and was forced removed the library and used native input of type="datetime-local", not so beautiful, but works.

**Comment by dbusacca** [comment]
I've had this Issue:
I the package.json of the project I've had:
`    "@angular/animations": "~6.0.3",`
`    "@angular/cdk": "~6.2.0",`
ng-pick-datetime requires ` @angular/cdk": "^6.0.0",`

Because of this a differente copy of @angular/cdk was pullend on the "private" node_module of ng-pick-datetime

I've resolved setting the my project dependency to "@angular/cdk": "^6.0.0"

Still i think that @angular/cdk should be declared as peer dependency

**Comment by OmkarJadhav21** [comment]
But  I **dont want to change angular cli or angular version**.Also by updating angular cdk version 6.0 **doesn't solve** my problem

**Comment by viethien** [comment]
I faced same issue when build at --prod mode.
@DanielYKPan  did you fix this issue for version 5.2.6 and 5.2.8. We can't upgrade angular to new version.
Can you provide step by step to fix this issue with angular 5.2.8?

**Comment by mbektas7** [comment]
I have faced to same issue when build with   `ng build --prod`.  I have updated all packages, ng datepicker and cli version. But never resolved. And then I have edited my angular.json file 
Aot: false
buildoptimize: false
and now its working.  I don't know that why I have faced to this issue. I donw want to disable aot or buildoptimize but now it's necessary. Maybe I'll change my datepicker library if issue is not resolve.

regards.
